### PR TITLE
[BugFix] Fix backend start script

### PR DESCRIPTION
**Pull request by dirtysalt**
## What type of PR is this：
- [X] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #https://github.com/StarRocks/starrocks/issues/16560

## Problem Summary(Required) ：

Some critical conf is missing in `start_cn.sh`, which makes CN can not load `java-extension`.

```
# HADOOP_CLASSPATH defined in $STARROCKS_HOME/conf/hadoop_env.sh
# put $STARROCKS_HOME/conf ahead of $HADOOP_CLASSPATH so that custom config can replace the config in $HADOOP_CLASSPATH
export CLASSPATH=$STARROCKS_HOME/conf:$STARROCKS_HOME/lib/jni-packages/*:$HADOOP_CLASSPATH:$CLASSPATH
```

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Review by stdpain — DISMISSED** [review]

File: bin/start_backend.sh [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+fi
+
+if [ "$JAVA_HOME" = "" ]; then
+    export LD_LIBRARY_PATH=$STARROCKS_HOME/lib/jvm/$jvm_arch/server:$STARROCKS_HOME/lib/jvm/$jvm_arch:$LD_LIBRARY_PATH
+else
+    java_version=$(jdk_version)
+    if [[ $java_version -gt 8 ]]; then
+        export LD_LIBRARY_PATH=$JAVA_HOME/lib/server:$JAVA_HOME/lib:$LD_LIBRARY_PATH
+        # JAVA_HOME is jdk
+        elif [[ -d "$JAVA_HOME/jre"  ]]; then
```
Comment (kevincai) [inline bin/start_backend.sh:94]: looks like incorrect indention.

Comment (dirtysalt) [reply bin/start_backend.sh:94]: thansk. good catch!

**Review by stdpain — APPROVED** [review]

**Review by Youngwb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by dirtysalt** [comment]
@MergifyIO backport branch-2.5 branch-2.4 branch-2.3 branch-2.2

**Pull request merged**